src/agent/onefuzz-task/src/tasks/utils.rs

```diff
@@ -96,3 +96,157 @@ pub fn parse_key_value(value: String) -> Result<(String, String)> {
 pub fn default_bool_true() -> bool {
     true
 }
+
+/// Try to resolve an ambiguous setup-relative subpath, returning an error if not found.
+pub async fn try_resolve_setup_relative_path(
+    setup_dir: impl AsRef<Path>,
+    subpath: impl AsRef<Path>,
+) -> Result<PathBuf> {
+    let setup_dir = setup_dir.as_ref();
+    let subpath = subpath.as_ref();
+
+    resolve_setup_relative_path(setup_dir, subpath)
+        .await?
+        .ok_or_else(|| {
+            anyhow::format_err!(
+                "unable to resolve subpath `{}` under setup dir `{}`",
+                subpath.display(),
+                setup_dir.display()
+            )
+        })
+}
+
+/// Try to resolve an ambiguous setup-relative subpath, returning `None` if not found.
+pub async fn resolve_setup_relative_path(
```

Comment:
@tevoinea, let me know if you have any questions about this behavior.

One thing I just noticed: I think a (valid, existing) absolute path will result in `Ok(None)`. We should probably check for that.

Reply:
(that is: we should either reject any absolute `subpath`, or alternatively, if `subpath` is absolute, and `setup_dir` is already a prefix of `subpath`, then this should arguably succeed)

Reply:
What do you think of logging a warning if more than one of these resolve? I imagine it's _probably_ a misconfiguration and the user would want to know.

Reply:
I'm going to stick with "return the first resolved" for now, because I don't actually want to actually attempt every single case. The main goal here is to have have a comprehensible resolution rule/order to aid debugging.